### ISSUE TITLE
mainnet, master, electrs dies

### ISSUE THREAD

**Issue by greenaddress** [romanz/electrs#22]
mainnet, full node synchronized and electrs synchronized

After a while the server dies because of some probably related to https://github.com/romanz/electrs/issues/18

```
2018-07-15T21:56:23.362+00:00 - WARN - no mempool entry 2b85b3dfdfb0f95644a49a154c720135effcee26d8b5c658376f43d7fd8dd695: missing fees section
2018-07-15T21:56:23.362+00:00 - ERROR - server failed: Error: non-array response
stack backtrace:
   0:     0x5618e5e05dac - backtrace::backtrace::trace::h44f27ad6608f02da
   1:     0x5618e5e05402 - backtrace::capture::Backtrace::new_unresolved::hb555da2dec445946
   2:     0x5618e5e04cd7 - error_chain::backtrace::imp::InternalBacktrace::new::hc1c04021afee31b3
   3:     0x5618e5e05044 - <error_chain::State as core::default::Default>::default::he24c386d1d298d64
   4:     0x5618e5d4187e - electrs::daemon::Daemon::requests::h9854bd24136c8b73
   5:     0x5618e5d43eed - electrs::daemon::Daemon::gettransactions::h93b25cda33204a41
   6:     0x5618e5d82ce0 - electrs::mempool::Tracker::update::h9fe07ffc4b13e9f1
   7:     0x5618e5d6002b - electrs::query::Query::update_mempool::h5a3567190cf4c2bb
   8:     0x5618e5d31df6 - electrs::main::h5294e30fc465d8cf
   9:     0x5618e5d30782 - std::rt::lang_start::{{closure}}::hc74e7e3dcbd83f22
  10:     0x5618e61e24b2 - std::rt::lang_start_internal::{{closure}}::h72602870b4a34363
                        at libstd/rt.rs:59
                         - std::panicking::try::do_call::hc2917ab2c572e22d
                        at libstd/panicking.rs:310
  11:     0x5618e6200469 - __rust_maybe_catch_panic
                        at libpanic_unwind/lib.rs:105
  12:     0x5618e61e99a5 - std::panicking::try::h3777f1f99a655493
                        at libstd/panicking.rs:289
                         - std::panic::catch_unwind::hab6d1467193fc938
                        at libstd/panic.rs:374
                         - std::rt::lang_start_internal::h106bcdfb2e3beaf8
                        at libstd/rt.rs:58
  13:     0x5618e5d32553 - main
  14:     0x7f4d433072b0 - __libc_start_main
  15:     0x5618e5d2a619 - _start
  16:                0x0 - <unknown>

```

**Comment by romanz** [comment]
Thanks for reporting this issue!
I'll add a log statement to better understand this issue.

**Comment by greenaddress** [comment]
@romanz thanks. Restarted with latest master

**Comment by gabridome** [comment]
FWIW I got exactly the same error.

**Comment by romanz** [comment]
@greenaddress @gabridome Thanks for helping reproducing the issue!

Could you please try the following version https://github.com/romanz/electrs/commit/57eb81e15d00c89f347b09f0c73b121e2a765369?
It should only log this error, since it should be transient (`getrawmempool()` and `getrawtransaction()` can get out of sync after a block is found / transaction is replaced by RBF exactly between these RPCs).

**Comment by gabridome** [comment]
This for the moment, was with the merge with 57eb81e:
````
2018-07-16T12:46:27.946+02:00 - WARN - no mempool entry 10d705eb56478c925b61e9bb87f6b3b355302dea64b9f2b1396ba5f4c18518e3: missing fees section
2018-07-16T12:46:27.947+02:00 - WARN - failed to get transactions []: non-array replies: Object({"error": Null, "id": Null, "result": Object({"ancestorcount": Number(1), "ancestorfees": Number(10000), "ancestorsize": Number(374), "depends": Array([]), "descendantcount": Number(1), "descendantfees": Number(10000), "descendantsize": Number(374), "fee": Number(0.0001), "height": Number(532142), "modifiedfee": Number(0.0001), "size": Number(374), "time": Number(1531737881), "wtxid": String("24a8d84ed60783207937d80e4ce37bc2a5ba833e7842047f393c592fc37f853d")})})
2018-07-16T12:46:32.947+02:00 - ERROR - server failed: Error: invalid blockhash
Caused by: non-string value
stack backtrace:
   0:     0x55587b39030c - backtrace::backtrace::trace::hd402417859783b40
   1:     0x55587b38f962 - backtrace::capture::Backtrace::new_unresolved::hce1354e135c2f28e
   2:     0x55587b38f237 - error_chain::backtrace::imp::InternalBacktrace::new::habdd565038f6f85c
   3:     0x55587b38f5a4 - <error_chain::State as core::default::Default>::default::hee976682d00eb04c
   4:     0x55587b3564f5 - electrs::daemon::parse_hash::h93f9be35b39a7f95
   5:     0x55587b35b01e - electrs::daemon::Daemon::getbestblockhash::h897efc5c99f6fa52
   6:     0x55587b333bd6 - electrs::app::App::update::h9aa7658f54799234
   7:     0x55587b2b7c93 - electrs::main::hb7893ebc8a38666e
   8:     0x55587b2bcf52 - std::rt::lang_start::{{closure}}::h825937f5b6ee0646
   9:     0x55587b7c12a2 - std::rt::lang_start_internal::{{closure}}::h342eaeec3a520255
                        at libstd/rt.rs:59
                         - std::panicking::try::do_call::h8b090b6144a018fb
                        at libstd/panicking.rs:310
  10:     0x55587b7df209 - __rust_maybe_catch_panic
                        at libpanic_unwind/lib.rs:105
  11:     0x55587b7c8795 - std::panicking::try::h11349d7f0dce3b3c
                        at libstd/panicking.rs:289
                         - std::panic::catch_unwind::hfbb0f172d1ed4df2
                        at libstd/panic.rs:374
                         - std::rt::lang_start_internal::h6c73689767a9401e
                        at libstd/rt.rs:58
  12:     0x55587b2b8463 - main
  13:     0x7f2a11ff382f - __libc_start_main
  14:     0x55587b2b5af8 - _start
  15:                0x0 - <unknown>

`````

**Comment by romanz** [comment]
Let's investigate this bug @ https://github.com/romanz/electrs/issues/24.

**Comment by romanz** [comment]
@gabridome Feel free to join our debugging session on https://gitter.im/romanz/electrs :)

**Comment by romanz** [comment]
Should be resolved together with #24
